### ISSUE TITLE
Any plan of migrating to function components?

### ISSUE THREAD

**Issue by llc1123** [Norserium/react-indiana-drag-scroll#65]
It's 2022 and I think we should refactor the code with function components.

Also react 18 is released and it should be compatible.

I'll consider making a PR if you think it's a good idea.

**Comment by Norserium** [comment]
@llc1123, hello! 

The component is elementary, so it's possible to rewrite it to a function component. But what's the real benefits? Currently I see only one advantage: we can extract the logic into the hook, that can be reused later.

**Comment by llc1123** [comment]
You are right. It is not necessary. But I think you should at least update `package.json` as it supports react 18.

**Comment by Norserium** [comment]
@llc1123, I've updated it in `2.2.0`. Try to update to that version.

**Comment by Norserium** [comment]
@llc1123, I've migrated to function components in `3.x.x`. There is [the example](https://norserium.github.io/react-indiana-drag-scroll/docs/api/use-scroll-container).
